ddtrace/internal/telemetry/writer.py

```diff
@@ -508,6 +511,33 @@ def _app_integrations_changed_event(self, integrations):
         }
         self.add_event(payload, "app-integrations-change")
 
+    def _app_client_configuration_changed_event(self, configuration):
+        # type: (List[Dict]) -> None
+        """Adds a Telemetry event which sends list of modified configurations to the agent"""
+
+        config_change_queue = []
+        # Loop though the events queue and find the app-started event payload
+        for e in self._events_queue:
+            if e["request_type"] == "app-started":
+                payload = e["payload"]
+                break
+        # Convert the original configuration to a dic to avoid iterating the whole list
+        original_config = {}
+        for cfg in self._configuration_queue:
+            original_config[cfg["name"]] = cfg["value"]
+        """
+        Find the list of configuration values to be changed.
+        If the input configuration doesn't collected by the app start event, it will be ignored
+        """
+        for changed_config in configuration:
+            cfg_name = changed_config["name"]
+            if cfg_name in original_config and original_config[cfg_name] != changed_config["value"]:
+                config_change_queue.append(changed_config)
+
+        # Only return configurations that being modified
```

Comment:
```suggestion
        # Only return configurations that have been modified
```
Putting lines 532-538 into a function `get_modified_configurations` would make the code a bit more readable and modular.